### DISCUSSION TITLE
Will fit_valid() be executed in data parallelism when using torch.distributed.launch --distributed_launch?

### DISCUSSION THREAD

**Discussion by My-captain** [speechbrain/speechbrain#1973] [Q&A]
I found that the valid_set is not decorated by `DistributedSamplerWrapper` in core.py

**Comment by TParcollet** [comment]
Hi, unfortunately, when using DDP all GPUs are evaluation with the full set. This was done for a reason, maybe @Gastron remembers?

**Comment by Gastron** [comment]
The validation computations could be parallelized in DDP, but it would require special code for passing the results from each process to the main node. It's not trivial how to do this automatically.

In specific cases where the validation takes a long time to compute, it may make sense to implement it for your specific validation metrics.
